### DISCUSSION TITLE
How to hide barbecue when opening diffview tab

### DISCUSSION THREAD

**Discussion by pirey** [utilyre/barbecue.nvim#126] [Q&A]
Hi.
I'm using https://github.com/sindrets/diffview.nvim for viewing git diff.
The problem is when opening staged changes in diffview, the git file (left) does not have barbecue while the other (right) file have it.
This makes the cursor does not align properly.
![image](https://github.com/user-attachments/assets/58ad05fa-a02d-4778-b57a-4a37c765db39)

I know there is an option called `exclude_filetypes`, but in this case it seems I need some kind of hook to trigger `:Barbecue hide` when I'm opening `:DiffviewOpen`

Is there anyway to workaround this?
Thanks

**Comment by h0adp0re** [comment]
Hey @pirey, this will do what you want:

```lua
vim.api.nvim_create_autocmd("User", {
  desc = "Hide barbecue when entering DiffView",
  pattern = "DiffviewViewEnter",
  callback = function()
    require("barbecue.ui").toggle(false)
  end,
})

vim.api.nvim_create_autocmd("User", {
  desc = "Show barbecue when leaving DiffView",
  pattern = "DiffviewViewLeave",
  callback = function()
    require("barbecue.ui").toggle(true)
  end,
})
```
